mart/nn/nn.py

```diff
@@ -308,3 +316,28 @@ def forward(self, *values, weights=None):
 
         assert len(weights) == len(values)
         return sum(value * weight for value, weight in zip(values, weights))
+
+
+# FIXME: This must exist already?!
+class TotalVariation(Sum):
+    def forward(self, *values, weights=None):
+        values = [self._total_variation(value) for value in values]
+
+        return super().forward(*values, weights=weights)
+
+    def _total_variation(self, image):
+        return torch.mean(
```

Comment:
Mean doesn't do anything and we should probably normalize this by the size of the patch (e.g., C*H*W).